core/src/wsv.rs

```diff
@@ -198,6 +212,18 @@ impl<W: WorldTrait> WorldStateView<W> {
         self.new_block_notifier.subscribe()
     }
 
+    fn produce_events(&self, events: impl IntoIterator<Item = DataEvent>) {
+        let events = events.into_iter().map(Event::from);
+        let events_sender = if let Some(sender) = &self.events_sender {
+            sender
+        } else {
+            return warn!("wsv does not equip an events sender");
+        };
+        for event in events {
+            drop(events_sender.send(event))
```

Comment:
why would you drop manually?

Reply:
I think in this case, the reason you want to drop manually has to do with the fact that the compiler can sometimes elide resource allocation. I'd try this particular snippet in compiler explorer to be sure, but it's not quite easy.

Reply:
This follows how Sumeragi send events.
Actually it does not work without explicit `drop`.
Yet I have not exactly understand why